### PR TITLE
Ocp cnv ipforwarding

### PR DESCRIPTION
**Pull request by doucol**
## Description
Implement ContainerIpForwarding option for CNI config.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/node_test.go [hunk 1]
```diff
@@ -986,6 +994,87 @@ var _ = Describe("Node rendering tests", func() {
 		Expect(GetContainer(ds.Spec.Template.Spec.InitContainers, "install-cni").VolumeMounts).To(ConsistOf(expectedCNIVolumeMounts))
 	})
 
+	It("should render a proper 'allow_ip_forwarding' container setting in the cni config[TestThis]", func() {
+		defaultInstance.Spec.FlexVolumePath = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
```
[the remark this answers is not in the document]
Comment (doucol) [reply pkg/render/node_test.go:998]: yes, agree - I had a difficult time getting working due to the json string comparison and this was left in there from a test that was actually working. but, agree we should take it out now.

File: pkg/render/node_test.go [hunk 2]
```diff
@@ -986,6 +994,87 @@ var _ = Describe("Node rendering tests", func() {
...
+      },
+      "kubernetes": {
+          "kubeconfig": "__KUBECONFIG_FILEPATH__"
+      }
+    },
+    {"type": "portmap", "snat": true, "capabilities": {"portMappings": true}}
+  ]
+}`))
+
+		dsResource := GetResource(resources, "calico-node", "calico-system", "apps", "v1", "DaemonSet")
```
[the remark this answers is not in the document]
Comment (doucol) [reply pkg/render/node_test.go:1038]: same comment as previous - also agree we should take these out now

File: pkg/apis/operator/v1/types.go [hunk 3]
```diff
@@ -206,6 +214,12 @@ type CalicoNetworkSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum=None;Multus
 	MultiInterfaceMode *MultiInterfaceMode `json:"multiInterfaceMode,omitempty"`
+
+	// ContainerIpForwarding configures whether ip forwarding will be enabled for containers in the CNI configuration.
+	// Default: Disabled
+	// +optional
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	ContainerIpForwarding *ContainerIpForwardingType `json:"containerIpForwarding,omitempty"`
```
Comment (caseydavenport) [inline pkg/apis/operator/v1/types.go:222]: Should be `containerIPForwarding` not `containerIpForwarding` to match conventional k8s API casing

Comment (caseydavenport) [reply pkg/apis/operator/v1/types.go:222]: Same for both the golang field name and the json tag

Comment (doucol) [reply pkg/apis/operator/v1/types.go]: done

**Pull request merged**